.github/workflows/kit-bootstrap.yml

```diff
@@ -0,0 +1,213 @@
+            kubectl --kubeconfig=$GUESTKUBECONFIG expose deployment nginx-deployment --name=nginx-service --port=80 --target-port=80 --type=ClusterIP
+            kubectl --kubeconfig=$GUESTKUBECONFIG certificate approve $(kubectl --kubeconfig=$GUESTKUBECONFIG get csr | grep Pending | awk '{print $1}')
+            kubectl --kubeconfig=$GUESTKUBECONFIG run -i --rm --restart=Never curly-pod  --image=curlimages/curl --command -- curl -s 'http://nginx-service:80'
+          done
+        working-directory: substrate
+      - name:  Delete the Guest Control plane
+        if: always()
+        run: | 
+              allVersions=("1-21" "1-22" "1-23" "1-24" "1-25")
+              if [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+                os_type="linux"    
+              elif [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+                os_type="darwin"
+              fi 
+              export KUBECONFIG=${HOME}/.kit/env/kit-github-${os_type}-${{ github.run_id }}-${{github.run_attempt}}/etc/kubernetes/admin.conf
+              kubectl config current-context
+              for version in "${allVersions[@]}"; do
+                GUEST_CLUSTER_NAME="guest-${os_type}-${version}-${{ github.run_id }}-${{github.run_attempt}}"
+                if kubectl get dataplane  $GUEST_CLUSTER_NAME-nodes > /dev/null 2>&1; then
+                  kubectl delete dataplane  $GUEST_CLUSTER_NAME-nodes
+                  echo "dataplane deleted."
+                else
+                  echo "No dataplane found  $GUEST_CLUSTER_NAME-nodes."
+                fi
+                if kubectl get controlplane  $GUEST_CLUSTER_NAME > /dev/null 2>&1; then
+                  kubectl delete controlplane  $GUEST_CLUSTER_NAME
+                  echo "Controlplane deleted."
+                else
+                  echo "No controlplane found  $GUEST_CLUSTER_NAME."
+                fi
+              done
+        continue-on-error: true
+        working-directory: substrate
+      - name:  Delete the kit environment 
+        if: always()
+        run: | 
+              #During the first attempt, "kitctl" was unable to delete some resources sporadically. To ensure that enough time is given for the resources to be deleted, a longer wait time of 220 seconds is added before attempting to delete the kit again.
+              #and for the remaining 4 iterations with a wait time of 30 seconds to ensure all resources are successfully deleted as we are left with few resources sometimes.
+              cd ${HOME}/work/kubernetes-iteration-toolkit/kubernetes-iteration-toolkit/substrate/cmd/kitctl
+              if [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+                os_type="linux"    
+              elif [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+                os_type="darwin"
+                /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+                brew install coreutils
```

Comment:
Why do we need this?
When you have something that's not obvious, you should add a comment.